packages/flutter_tools/lib/src/commands/ios_analyze.dart

```diff
@@ -0,0 +1,70 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import '../base/file_system.dart';
+import '../base/logger.dart';
+import '../convert.dart';
+import '../ios/xcodeproj.dart';
+import '../project.dart';
+
+/// The type of analysis to perform.
+enum IOSAnalyzeOption {
+  /// Prints out available build variants of the iOS Xcode sub-project.
+  ///
+  /// An example output:
+  ///
+  /// {"configurations":["Debug","Release","Profile"],"schemes":["Runner"],"targets":["Runner","RunnerTests"]}
+  listBuildOptions,
+
+  /// Outputs universal link settings of the iOS Xcode sub-project into a file.
+  ///
+  /// The file path will be printed after the command is run successfully.
+  outputUniversalLinkSettings,
+}
+
+/// Analyze the iOS Xcode sub-project of a Flutter project.
+///
+/// The [userPath] must be point to a flutter project.
+class IOSAnalyze {
+  IOSAnalyze({
+    required this.project,
+    required this.option,
+    this.configuration,
+    this.scheme,
+    this.target,
+    required this.logger,
+  }) : assert(option == IOSAnalyzeOption.listBuildOptions ||
+              (configuration != null && scheme != null && target != null));
+
+  final FlutterProject project;
+  final IOSAnalyzeOption option;
+  final String? configuration;
+  final String? scheme;
+  final String? target;
```

Comment:
These can all be non-nullable fields, right?

Reply:
no, in case of `--list-build-options` these can be null

Reply:
ahh, i see